src/actions/sources/select.js

```diff
@@ -128,7 +130,12 @@ export function selectLocation(
       source = getSourceFromId(getState(), location.sourceId);
     }
 
-    dispatch(addTab(source.url, 0));
+    dispatch(addTab(source.url));
+    const framework = getFramework(getTabs(getState()), source.url);
+    if (framework) {
+      dispatch(updateTab(source.url, framework));
```

Comment:
@AnshulMalik Could you describe the need for this update call?

Reply:
Just realised, that's not needed anymore :)